src/cosl/coordinated_workers/coordinator.py

```diff
@@ -370,6 +377,14 @@ def __init__(
             )
             return
 
+        if len(self.cluster.gather_workload_versions()) > 1:
+            logger.error(
+                f"Incoherent deployment. {charm.unit.name} is connected to workers that are running "
+                f"different workload versions: {', '.join(self.cluster.gather_workload_versions())}. "
```

Comment:
instead of calling that twice, lift it out to a variable or define it in a walrus